src/aubuf/aubuf.c

```diff
@@ -231,72 +304,46 @@ int aubuf_write_auframe(struct aubuf *ab, struct auframe *af)
  */
 void aubuf_read_auframe(struct aubuf *ab, struct auframe *af)
 {
-	struct le *le;
 	size_t sz;
-	size_t sample_size;
-	uint8_t *p;
 	bool filling;
+	enum ajb_state as;
 
 	if (!ab || !af)
 		return;
 
-	sample_size = aufmt_sample_size(af->fmt);
-	if (sample_size)
-		sz = af->sampc * sample_size;
-	else
-		sz = af->sampc;
-
-	p = af->sampv;
+	if (!ab->ajb && ab->mode == AUBUF_ADAPTIVE)
+		ab->ajb = ajb_alloc(ab->silence);
 
 	lock_write_get(ab->lock);
+	as = ajb_get(ab->ajb, af);
+	if (as == AJB_LOW) {
+		re_printf("aubuf: inc buffer due to high jitter\n");
```

Comment:
We should avoid re_printf (its blocking, when fflush it can block for multiple ms) or only with AUBUF_DEBUG.